frontend/src/pages/integrations/gcp-secret-manager/create.tsx

```diff
@@ -0,0 +1,127 @@
+  const { data: integrationAuthApps } = useGetIntegrationAuthApps({ integrationAuthId: integrationAuthId as string ?? '' });
+
+  const [selectedSourceEnvironment, setSelectedSourceEnvironment] = useState('');
+  const [owner, setOwner] = useState<string | null>(null);
+  const [targetApp, setTargetApp] = useState('');
+  const [targetAppId, setTargetAppId] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (workspace) {
+      setSelectedSourceEnvironment(workspace.environments[0].slug);
+    }
+  }, [workspace]);
+
+  useEffect(() => {
+    // TODO: handle case where apps can be empty
+    if (integrationAuthApps) {
+      setTargetApp(integrationAuthApps[0].name);
+      setTargetAppId(integrationAuthApps[0].appId ?? '');
+      setOwner(integrationAuthApps[0]?.owner ?? null);
+    }
+  }, [integrationAuthApps]);
+
+  const handleButtonClick = async () => {
+    try {
+      setIsLoading(true);
+
+      if (!integrationAuth?._id) return;
+
+      await createIntegration({
+        integrationAuthId: integrationAuth?._id,
+        isActive: true,
+        app: targetApp,
+        appId: targetAppId,
+        sourceEnvironment: selectedSourceEnvironment,
+        targetEnvironment: null,
+        owner,
+        path: null,
+        region: null
+      });
+
+      setIsLoading(false);
+      router.push(
+        `/integrations/${localStorage.getItem('projectData.id')}`
+      );
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  return (integrationAuth && workspace && selectedSourceEnvironment && integrationAuthApps && targetApp) ? (
+    <div className="h-full w-full flex justify-center items-center">
+      <Card className="max-w-md p-8 rounded-md">
+        <CardTitle className='text-center'>GCP Secret Manager Integration</CardTitle>
+        <FormControl
+          label="Project Environment"
+          className='mt-4'
+        >
+          <Select
+            value={selectedSourceEnvironment}
+            onValueChange={(val) => setSelectedSourceEnvironment(val)}
+            className='w-full border border-mineshaft-500'
+          >
+            {workspace?.environments.map((sourceEnvironment) => (
+              <SelectItem value={sourceEnvironment.slug} key={`gcp-secret-manager-environment-${sourceEnvironment.slug}`}>
+                {sourceEnvironment.name}
+              </SelectItem>
+            ))}
+          </Select>
+        </FormControl>
+        <FormControl
+          label="GCP Projects"
+          className='mt-4'
+        >
+          <Select
+            value={targetApp}
+            onValueChange={(val) => setTargetApp(val)}
+            className='w-full border border-mineshaft-500'
+          >
+            {integrationAuthApps.map((integrationAuthApp) => (
+              <SelectItem value={integrationAuthApp.name} key={`gcp-secret-manager-environment-${integrationAuthApp.name}`}>
```

Comment:
You may have to do the same for value as well potentially since I'm receiving this kinda edge-case where I have multiple projects with the same name and selecting one, selects all of them.